.changeset/sixty-balloons-build.md

```diff
@@ -0,0 +1,31 @@
+---
+'@emotion/cache': minor
+'@emotion/sheet': minor
+---
+
+Add insertionPoint option to the EmotionCache, to insert rules after the specified element.
+
+```jsx
+const head = document.querySelector('head')
+
+// <meta name="emotion-insertion-point" content="">
+const emotionInsertionPoint = document.createElement('meta')
+emotionInsertionPoint.setAttribute('name', 'emotion-insertion-point')
+emotionInsertionPoint.setAttribute('content', '')
```

Comment:
FYI: I've checked the spec and it seems that `content` attribute is required in this case:
> If either name, http-equiv, or itemprop is specified, then the content attribute must also be specified. Otherwise, it must be omitted.

Reply:
Kindly, does this work with SSR, 'cause `document` seems to be undefined in my case? What do you suggest I do?
Thanks.

Reply:
You need to create such a tag (or similar) in your SSR response and then select it on the client's side and pass it as the `insertionPoint` to `createCache`

Reply:
Hi, @Andarist. Thanks for your response. I saw and did as you suggested and it worked. Thank you.

But then, there's another issue which I'm noticing. I'm using Framer Motion's page transition feature--AnimatePresence--with MUI, and it happens that on route change start, the styles of the unmounting page/component gets removed from the DOM causing a FOUC of the unmounting page (before it transitions out).

Is there a way to preserve the styles of the pages as they get mounted, so they don't have to get removed and re-added everytime they get mounted/unmounted. I hope you understand what I mean?

Reply:
What styles are removed on the route change? By default Emotion doesn't unmount any styles - except global ones when used through `<Global/>`. Framer Motion also works in a way that it temporarily preserve components that are unmounted in the tree (they just render exiting children for longer than you are rendering them - they can do that cause they just cache previously received children). So even if some styles would get unmounted during the process they should be unmounted after the component goes completely out of the page so there should be no FOUC.

So while I understand what you are observing there, I don't understand how this can happen. Do you have a repro case (even a complex one, like a full deployed page) that I could take a look at?

Reply:
Thanks for your response(s) again. I do appreciate them.

Ok. I don't have a deployed page yet of the app, as it's not fully ready to be deployed, but I can make a short video and send. Not sure if GitHub accepts accepts videos in the comments.

I'll try or probably send a link.

I'll do that in a moment.

Thank you.

Reply:
What it's supposed to look like...
https://user-images.githubusercontent.com/28790485/152666236-7758fccd-d279-46e4-bc16-6ff7472d4cc7.mov

Reply:
...versus what it currently looks like in prod:

https://user-images.githubusercontent.com/28790485/152666255-2a1b4684-bfb5-4c2e-96ad-bed96d099bbe.mov

:

Reply:
@Andarist , so you'll notice the FOUC in the production (build) video.

Reply:
Hi, @Andarist . Seen this yet?

Reply:
Could you perhaps record this with https://www.replay.io/ ? I'd like to debug this and can't do that with just a standard recording of the problem.

Reply:
Alright, I will. Thanks for your support. 🤲🏼